packages/core/src/lib/lit-core.ts

```diff
@@ -414,212 +566,146 @@ export class LitCore {
+        Object.values(serverKeys).map(
+          (keysFromSingleNode: any) => keysFromSingleNode.hdRootPubkeys
+        )
+      ),
+      latestBlockhash,
+      lastBlockHashRetrieved: Date.now(),
+    } as CoreNodeConfig;
+  }
 
-          // browser only
-          if (isBrowser()) {
-            document.dispatchEvent(new Event('lit-ready'));
-          }
-          // if the interval is defined we clear it
-          if (this.networkSyncInterval) {
-            clearInterval(this.networkSyncInterval);
-          }
+  /** Currently, we perform a full sync every 30s, including handshaking with every node
+   * However, we also have a state change listener that watches for staking contract state change events, which
+   * _should_ be the only time that we need to perform handshakes with every node.
+   *
+   * However, the current block hash does need to be updated regularly, and we currently update it only when we
+   * handshake with every node.
+   *
+   * We can remove this network sync code entirely if we refactor our code to fetch latest blockhash on-demand.
+   * @private
+   */
+  private scheduleNetworkSync() {
+    if (this._networkSyncInterval) {
+      clearInterval(this._networkSyncInterval);
+    }
 
-          this.networkSyncInterval = setInterval(async () => {
-            if (Date.now() - this.lastBlockHashRetrieved! >= 30_000) {
-              log(
-                'Syncing state for new network context current config: ',
-                this.config,
-                'current blockhash: ',
-                this.lastBlockHashRetrieved
-              );
-              await this._runHandshakeWithBootstrapUrls().catch((err) => {
-                throw err;
-              });
-              log(
-                'Done syncing state new config: ',
-                this.config,
-                'new blockhash: ',
-                this.lastBlockHashRetrieved
-              );
-            }
-          }, 30_000);
-
-          // @ts-ignore: Expected 1 arguments, but got 0. Did you forget to include 'void' in your type argument to 'Promise'?ts(2794)
-          resolve();
-        } else {
-          const now = Date.now();
-          if (now - startTime > this.config.connectTimeout) {
-            clearInterval(interval);
-            const msg = `Error: Could not connect to enough nodes after timeout of ${
-              this.config.connectTimeout
-            }ms.  Could only connect to ${
-              Object.keys(this.serverKeys).length
-            } of ${
-              this.config.bootstrapUrls.length
-            } required nodes.  Please check your network connection and try again.  Note that you can control this timeout with the connectTimeout config option which takes milliseconds.`;
-            logErrorWithRequestId(requestId, msg);
-            reject(msg);
-          }
+    this._networkSyncInterval = setInterval(async () => {
+      if (Date.now() - this.lastBlockHashRetrieved! >= 30_000) {
```

Comment:
can we move this `30_000` as constant at the top of the file?

Reply:
:+1: done!